test/widgets/action_sheet_test.dart

```diff
@@ -135,31 +168,42 @@ void main() {
       await tester.longPress(find.text(topic));
       // sheet appears onscreen; default duration of bottom-sheet enter animation
       await tester.pump(const Duration(milliseconds: 250));
-      check(find.byType(BottomSheet)).findsOne();
-      check(find.text('Follow topic')).findsOne();
-    });
+    }
+
+    Future<void> showFromAppBar(WidgetTester tester, {
+      ZulipStream? channel,
+      String topic = someTopic,
+      Iterable<StreamMessage>? messages,
+    }) async {
+      final effectiveChannel = channel ?? someChannel;
+      final effectiveMessages = messages ?? [someMessage];
+      assert(effectiveMessages.every((m) => m.topic.apiName == topic));
 
-    testWidgets('show from app bar', (tester) async {
-      await prepare();
       connection.prepare(json: eg.newestGetMessagesResult(
-        foundOldest: true, messages: [message]).toJson());
+        foundOldest: true, messages: effectiveMessages.toList()).toJson());
       await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
         child: MessageListPage(
-          initNarrow: eg.topicNarrow(channel.streamId, topic))));
+          initNarrow: eg.topicNarrow(effectiveChannel.streamId, topic))));
       // global store, per-account store, and message list get loaded
       await tester.pumpAndSettle();
 
-      await tester.longPress(find.byType(ZulipAppBar));
+      final topicRow = find.descendant(
+        of: find.byType(ZulipAppBar),
+        matching: find.text(topic));
+      await tester.longPress(topicRow);
       // sheet appears onscreen; default duration of bottom-sheet enter animation
       await tester.pump(const Duration(milliseconds: 250));
-      check(find.byType(BottomSheet)).findsOne();
-      check(find.text('Follow topic')).findsOne();
-    });
+    }
+
+    Future<void> showFromRecipientHeader(WidgetTester tester, {
+      String topic = someTopic,
+      StreamMessage? message,
+    }) async {
+      final effectiveMessage = message ?? someMessage;
+      assert(effectiveMessage.topic.apiName == topic);
```

Comment:
Can this function skip taking `topic` as a parameter, and just get it from the message?